### PR TITLE
Bigtable (#83)

### PR DESCRIPTION
**Pull request by ccemeraldeyes**

### CODE REVIEW

File: brokerapi/brokers/bigtable/broker.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+		Url:          "",
+		Location:     "",
+		OtherDetails: string(otherDetails),
+	}
+
+	return i, nil
+}
+
+// deletes the instance associated with the given instanceID string
+// note that all tables in the dataset must be deleted prior to deprovisioning TODO: still true?
```
Comment (johnsonj) [inline brokerapi/brokers/bigtable/broker.go]: file a bug for this todo

File: brokerapi/brokers/gcp_service_broker.go [hunk 2]
```diff
@@ -561,6 +580,70 @@ func InitCatalogFromEnv() ([]models.Service, error) {
...
+	var btDynamicPlans map[string]BigtableDynamicPlan
+	btDynamicPlanJson := os.Getenv("BIGTABLE_CUSTOM_PLANS")
+
+	if btDynamicPlanJson != "" {
+		err = json.Unmarshal([]byte(btDynamicPlanJson), &btDynamicPlans)
+		if err != nil {
+			return []models.Service{}, fmt.Errorf("Error unmarshalling bigtable custom plan json %s", err)
+		}
+
+		// save cloudsql plans to database and construct mapping
```
Comment (johnsonj) [inline brokerapi/brokers/gcp_service_broker.go]: bigtable

File: brokerapi/brokers/gcp_service_broker.go [hunk 3]
```diff
@@ -561,6 +580,70 @@ func InitCatalogFromEnv() ([]models.Service, error) {
...
+				return []models.Service{}, err
+			}
+
+			id, err := db_service.GetOrCreatePlanId(planName, planDetails.ServiceId)
+			if err != nil {
+				return []models.Service{}, err
+			}
+			currentPlanIds = append(currentPlanIds, id)
+
+			features := map[string]string{
```
Comment (johnsonj) [inline brokerapi/brokers/gcp_service_broker.go]: can you share this code with CloudSql? seems like building this features map and the metadata are the only unique pieces.

File: integration_tests/integration_test.go [hunk 4]
```diff
@@ -139,9 +140,13 @@ func testGenericService(gcpBroker *GCPAsyncServiceBroker, params *genericService
 //   not fail if the resource existed before hand.
 // - If we always use a static one, globally named resources (eg, a storage bucket)
 //   would fail to create when two different projects run these tests.
-func generateInstanceName(projectId string) string {
+func generateInstanceName(projectId string, sep string) string {
 	hashed := crc32.ChecksumIEEE([]byte(projectId))
-	return fmt.Sprintf("pcf_sb_1_%d", hashed)
+	if sep != "" {
```
Comment (johnsonj) [inline integration_tests/integration_test.go]: another option here is:
```go
if sep == "" {
 sep = "_"
}
return ...
```

File: integration_tests/integration_test.go [hunk 5]
```diff
@@ -297,12 +322,24 @@ var _ = Describe("LiveIntegrationTests", func() {
...
+			}
+		}`)
+
 		var creds models.GCPCredentials
 		creds, err = brokers.GetCredentialsFromEnv()
 		if err != nil {
 			logger.Error("error", err)
 		}
-		instance_name = generateInstanceName(creds.ProjectId)
+		instance_name = generateInstanceName(creds.ProjectId, "")
```
Comment (johnsonj) [inline integration_tests/integration_test.go]: or perhaps better yet just pass "_" here

File: integration_tests/integration_test.go [hunk 6]
```diff
@@ -377,117 +414,48 @@ var _ = Describe("LiveIntegrationTests", func() {
...
+		BeforeEach(func() {
+			bigtableInstanceName = generateInstanceName(gcpBroker.RootGCPCredentials.ProjectId, "-")
+			name_generator.Basic = &fakes.StaticNameGenerator{Val: bigtableInstanceName}
+		})
+
+		AfterEach(func() {
+			name_generator.Basic = &fakes.StaticNameGenerator{Val: instance_name}
+		})
+
+		It("can provision/bind/unbind/deprovision", func() {
```
Comment (johnsonj) [inline integration_tests/integration_test.go]: :+1:

File: brokerapi/brokers/gcp_service_broker.go (hunk 3 above)
Comment (ccemeraldeyes) [inline brokerapi/brokers/gcp_service_broker.go]: Agreed. Going to address this as a separate refactoring commit on develop

**Pull request merged**